lib/cadet_web/controllers/incentives_controller.ex

```diff
@@ -2,20 +2,40 @@ defmodule CadetWeb.IncentivesController do
   use CadetWeb, :controller
 
   use PhoenixSwagger
+  import Cadet.Assessments
 
+  alias Cadet.Accounts.CourseRegistrations
   alias Cadet.Incentives.{Achievements, Goals}
 
   def index_achievements(conn, _) do
     course_id = conn.assigns.course_reg.course_id
     render(conn, "index_achievements.json", achievements: Achievements.get(course_id))
   end
 
+  @spec index_goals(Plug.Conn.t(), any) :: Plug.Conn.t()
   def index_goals(conn, _) do
     render(conn, "index_goals_with_progress.json",
       goals: Goals.get_with_progress(conn.assigns.course_reg)
     )
   end
 
+  def calculate_totalxp(conn, _) do
+    course_id = conn.assigns.course_reg.course_id
+    total_achievement_xp = Achievements.get_total_xp(course_id)
+    user = conn.assigns.current_user
+
+    if user.latest_viewed_course_id do
```

Comment:
You should not access `latest_viewed_course_id` here, because the route specifies a specific course ID.

What you probably want is `conn.assigns.course_reg.id`...